File: CsLua/TypeConverter.cs
namespace CsLua
{
    public static class TypeConverter
    {
        public static object EnsureTuple(object obj)
        {
            return obj;
        }

        public static MultipleArguments MultipleArguments(object o1)
        {
            return new MultipleArguments();
        }

        public static MultipleArguments MultipleArguments(object o1, object o2)
        {
            return new MultipleArguments();
        }

        public static MultipleArguments MultipleArguments(object o1, object o2, object o3)
        {
            return new MultipleArguments();
        }
    }
}

File: CsLua/TypeConverter.lua
--TargetFile: CsLua.lua

CsLua = CsLua or {};
CsLua.TypeConverter = {
    EnsureTuple = function(...)
        local t = {...};
        if #(t) == 1 and type(t[1]) == "table" and t[1].__type == 'Tuple' then
            return t[1];
        end
        return CsLua.Tuple(...);
    end,
    MultipleArguments = function(...)
        return ...;
    end
};